src/monkeyflocker/client.py

```diff
@@ -45,17 +81,147 @@ def _make_userlist(
+                shadow_requests.append(
+                    ShadowRequest(
+                        url=f"{url}/log",
+                        username=u.name,
+                        method="GET",
+                        data=None,
+                    )
+                )
+                requests.append(client.get(url=url))
+                shadow_requests.append(
+                    ShadowRequest(
+                        url=url, username=u.name, method="GET", data=None
+                    )
+                )
+
+        if command == "stop":
+            # We want to stack all the deletes at the end, rather than
+            # interleaving
+            for u in self.users:
+                requests.append(client.delete(url=f"{self.endpoint}/{u.name}"))
+                shadow_requests.append(
+                    ShadowRequest(
+                        url=url, username=u.name, method="DELETE", data=None
+                    )
+                )
+
+        output = await self._retry_requests(client, requests, shadow_requests)
+        if command == "report" or command == "stop":
+            await self._generate_output(output)
         await client.close()
+
+    async def _retry_requests(
+        self,
+        client: aiohttp.ClientSession,
+        requests: List[Any],
+        shadow_requests: List[ShadowRequest],
+    ) -> Dict[str, str]:
+        returned_text: Dict[str, str] = {}
+        delay: int = 1
+        max_delay: int = 30
+        max_tries: int = 10
+        count = 0
+        while count < max_tries:
+            if count != 0:
+                self.log.info(
+                    f"Waiting {delay}s before retrying failed requests"
+                    + f" {count}/{max_tries}"
+                )
+                await asyncio.sleep(delay)
+                delay *= 2
+                if delay > max_delay:
+                    delay = max_delay
+            count += 1
+            results = []
+            try:
+                results = await asyncio.gather(
+                    *requests, return_exceptions=True
+                )
+            except TypeError as exc:
+                self.log.error(
+                    "TypeError trying to gather something vile:" + f" {exc}."
+                )
+            retry = []
+            for idx, res in enumerate(results):
+                shadow = shadow_requests[idx]
+                method = shadow.method
+                url = shadow.url
+                data = shadow.data
+                username = shadow.username
+                if isinstance(res, Exception):
+                    # It's not always a ClientResponseError
+                    status: int = -1
+                    if isinstance(res, aiohttp.ClientResponseError):
+                        status = res.status
+                    if (status != 404) or (method not in ["DELETE", "GET"]):
+                        # 404 from DELETE or GET is OK-ish
+                        # Probably means "already deleted"
```

Comment:
I think it's clearer to write this sort of thing like:
```
if status == 404 or method in ("DELETE", "GET"):
    # 404 from DELETE or GET is OK-ish
    # Probably means "already deleted"
    continue
```
and then move all the following code outside the conditional to make it clear that we're just doing nothing in this specific case.

Reply:
Changed massively, but short answer is, we can't "continue" because we need to do something at the end of the loop.  At least in the newer version.